pkg-manager/plugin-commands-installation/src/recursive.ts

```diff
@@ -363,7 +364,7 @@ export async function recursive (
           {
             ...installOpts,
             ...localConfig,
-            ...getOptionsFromRootManifest(manifest),
+            ...getOptionsFromRootManifest(opts.rootProjectManifest ?? manifest),
```

Comment:
@zkochan so the patching error is due to the earlier proposed change (`opts.rootProjectManifest ?? {}` instead of `opts.rootProjectManifest ?? manifest` in the latest update).

In the latest update, I uses the project manifest as a fallback instead of an empty object.
Now the tests will pass as you can see those in mine:
https://github.com/alvis/pnpm/actions/runs/6338054374

The change is harmless but I guess there is a bug in the test because the earlier proposed logic is the same other places in the codebase, and it makes more sense than using the project manifest e.g.
https://github.com/pnpm/pnpm/blob/a5fff1cf51cbb1c0b680cb0714bb01df1bf7b2be/pkg-manager/plugin-commands-installation/src/link.ts#L178

https://github.com/pnpm/pnpm/blob/a5fff1cf51cbb1c0b680cb0714bb01df1bf7b2be/pkg-manager/plugin-commands-installation/src/remove.ts#L175

Regardless, I'm looking forward to your review and get it merged.